pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -0,0 +1,624 @@
+             # disabled type check due to too many xfails; see GH 23982 / 25425
+             # and type(result_fill_value) == type(expected_fill_value)
+             )
+            or (result_fill_value is np.nan and expected_fill_value is np.nan)
+            or (result_fill_value is NaT and expected_fill_value is NaT))
+
+
+def test_maybe_promote_int_with_int():
+    # placeholder due to too many xfails; see GH 23982 / 25425
+    pass
+
+
+@pytest.mark.parametrize('boxed, box_dtype', [
+    (True, None),    # fill_value wrapped in array with auto-dtype
+    # disabled due to too many xfails; see GH 23982 / 25425
+    # (True, object),  # fill_value wrapped in array with object dtype
+    (False, None)    # fill_value directly
+])
+def test_maybe_promote_int_with_float(any_int_dtype, float_dtype,
+                                      boxed, box_dtype):
+    dtype = np.dtype(any_int_dtype)
+    fill_dtype = np.dtype(float_dtype)
+
+    if float_dtype == 'float32' and not boxed:
+        pytest.xfail('falsely upcasts to float64')
+
+    # create array of given dtype; casts "1" to correct dtype
+    fill_value = np.array([1], dtype=fill_dtype)[0]
+
+    # filling int with float always upcasts to float64
+    expected_dtype = np.float64
+    # fill_value can be different float type
+    exp_val_for_scalar = np.float64(fill_value)
+    exp_val_for_array = np.nan
+
+    _check_promote(dtype, fill_value, boxed, box_dtype, expected_dtype,
+                   exp_val_for_scalar, exp_val_for_array)
+
+
+@pytest.mark.parametrize('boxed, box_dtype', [
+    (True, None),    # fill_value wrapped in array with auto-dtype
+    # disabled due to too many xfails; see GH 23982 / 25425
+    # (True, object),  # fill_value wrapped in array with object dtype
+    (False, None)    # fill_value directly
+])
+def test_maybe_promote_float_with_int(float_dtype, any_int_dtype,
+                                      boxed, box_dtype):
+
+    dtype = np.dtype(float_dtype)
+    fill_dtype = np.dtype(any_int_dtype)
+
+    # create array of given dtype; casts "1" to correct dtype
+    fill_value = np.array([1], dtype=fill_dtype)[0]
+
+    # filling float with int always keeps float dtype
+    # because: np.finfo('float32').max > np.iinfo('uint64').max
+    expected_dtype = dtype
+    # output is not a generic float, but corresponds to expected_dtype
+    exp_val_for_scalar = np.array([fill_value], dtype=expected_dtype)[0]
+    exp_val_for_array = np.nan
+
+    _check_promote(dtype, fill_value, boxed, box_dtype, expected_dtype,
+                   exp_val_for_scalar, exp_val_for_array)
+
+
+def test_maybe_promote_float_with_float():
+    # placeholder due to too many xfails; see GH 23982 / 25425
+    pass
+
+
+@pytest.mark.parametrize('boxed, box_dtype', [
+    (True, None),    # fill_value wrapped in array with auto-dtype
+    (True, object),  # fill_value wrapped in array with object dtype
+    (False, None)    # fill_value directly
```

Comment:
what does box_dtype of None even mean?

Reply:
I'm trying to explain just that with the comment next to it.

Reply:
Side note here: None takes no extra meaning, but is passed directly to the constructor of `np.array(some_value, dtype=None)`, which corresponds to the default behaviour of numpy.

Reply:
Incorporate this side-note as a comment.  That will help to explain the parameterized values better.